PyriteEditor/src/CornellBoxScene.h

```diff
@@ -0,0 +1,172 @@
+            m_RDG.addPass(&m_depthPrePass);
+            m_RDG.addPass(&m_SSAOPass);
+            m_RDG.addPass(&m_forwardPass);
+            m_RDG.getResourcesManager().addProduced(&m_depthPrePass, "depthBuffer");
+            m_RDG.getResourcesManager().addProduced(&m_SSAOPass, "ssaoTexture_blurred");
+            m_RDG.getResourcesManager().addProduced(&m_SSAOPass, "ssaoTexture");
+            m_RDG.getResourcesManager().addRequirement(&m_SSAOPass, "depthBuffer");
+            m_RDG.getResourcesManager().linkResource(&m_depthPrePass, "depthBuffer", &m_SSAOPass);
+            m_RDG.getResourcesManager().linkResource(&m_SSAOPass, "ssaoTexture_blurred", &m_forwardPass);
+            m_forwardPass.boundCamera = &m_camera;
+            bool bIsGraphValid = m_RDG.getResourcesManager().checkResourcesValidity();
+#pragma endregion RDG
+
+            sceneMeshes.reserve(m_cornellBoxModels.size());
+            for (const std::shared_ptr<pyr::Model>& model : m_cornellBoxModels)
+            {
+                sceneMeshes.emplace_back(pyr::StaticMesh{ model });
+                sceneMeshes.back().getTransform().scale = { 10,10,10 };
+                m_forwardPass.addMeshToPass(&sceneMeshes.back());
+                m_depthPrePass.addMeshToPass(&sceneMeshes.back());
+            }
+            m_lights.Points.push_back({});
+            m_lights.Points.push_back({});
+            m_lights.Directionals.push_back(pyr::DirectionalLight{});
+            m_lights.Spots.push_back({});
+            m_lights.Spots.push_back({});
+            m_lights.Spots.push_back({});
+
+            std::vector<pyr::hlsl_GenericLight> gpu_lights = m_lights.ConvertCollectionToHLSL();
+
+        }
+
+        /// ------------------------------------------------------------------------------------------------------------------------------- ///
+        
+        void update(float delta) override
+        {
+            m_camController.processUserInputs(delta);
+            pcameraBuffer->setData(CameraBuffer::data_t{
+                .mvp = m_camera.getViewProjectionMatrix(),
+                .pos = m_camera.getPosition()
+                });
+            pinvCameBuffer->setData(InverseCameraBuffer::data_t{
+                .inverseViewProj = m_camera.getViewProjectionMatrix().Invert(),
+                .inverseProj = m_camera.getProjectionMatrix().Invert(),
+                .Proj = m_camera.getProjectionMatrix()
+                });
+
+            LightsBuffer::data_t light_data{};
+            std::copy_n(m_lights.ConvertCollectionToHLSL().begin(), 16, std::begin(light_data.lights));
+            pLightBuffer->setData(light_data);
+        }
+
+        /// ------------------------------------------------------------------------------------------------------------------------------- ///
+        
+        void render() override
+
+        {
+            pyr::Engine::d3dcontext().IASetPrimitiveTopology(D3D11_PRIMITIVE_TOPOLOGY_TRIANGLELIST);
```

Comment:
Could become a part of the RenderProfiles at some point

Reply:
yeah but bothers me to do it so i'll do it later